kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-user-system/kie-server-services-task-assigning-user-system-simple/src/main/java/org/kie/server/services/taskassigning/user/system/simple/SimpleUserSystemServiceHelper.java

```diff
@@ -0,0 +1,211 @@
+
+        for (ElementLine line : lines) {
+            Set<Group> userGroups = new HashSet<>();
+            User user = new UserImpl(line.elementId, userGroups, new HashMap<>());
+            line.values.forEach(groupName -> {
+                Group group = groupMap.computeIfAbsent(groupName, GroupImpl::new);
+                userGroups.add(group);
+            });
+            usersMap.put(user.getId(), user);
+        }
+        populateAttribute(usersMap, SKILLS_ATTRIBUTE_NAME, skillLines);
+        populateAttribute(usersMap, AFFINITIES_ATTRIBUTE_NAME, affinityLines);
+        return new UserGroupInfo(new ArrayList<>(usersMap.values()), new ArrayList<>(groupMap.values()));
+    }
+
+    public static class UserGroupInfo {
+
+        private List<User> users;
+        private List<Group> groups;
+
+        UserGroupInfo(List<User> users, List<Group> groups) {
+            this.users = users;
+            this.groups = groups;
+        }
+
+        public List<User> getUsers() {
+            return users;
+        }
+
+        public List<Group> getGroups() {
+            return groups;
+        }
+    }
+
+    private static class ElementLine {
+
+        private String elementId;
+        private List<String> values = new ArrayList<>();
+
+        public ElementLine(String elementId) {
+            this.elementId = elementId;
+        }
+
+        public String getElementId() {
+            return elementId;
+        }
+
+        public void addValue(String value) {
+            values.add(value);
+        }
+
+        public List<String> getValues() {
+            return values;
+        }
+    }
+
+    private static void populateAttribute(Map<String, User> usersMap, String attributeName, List<ElementLine> attributeLines) {
+        for (ElementLine attributeLine : attributeLines) {
+            User user = usersMap.get(attributeLine.getElementId());
+            if (user != null) {
+                final String value = String.join(",", attributeLine.getValues());
+                if (value.length() > 1) {
+                    user.getAttributes().put(attributeName, value);
+                }
+            }
+        }
+    }
+
+    private static List<ElementLine> readLines(InputStream input) throws IOException {
+        final List<String> rawLines = IOUtils.readLines(input, StandardCharsets.UTF_8).stream()
+                .map(String::trim)
+                .filter(StringUtils::isNotEmpty)
+                .filter(line -> !line.startsWith("#"))
+                .collect(Collectors.toList());
+        final List<ElementLine> lines = new ArrayList<>();
+        for (String rawLine : rawLines) {
+            final String[] rawLineSplit = rawLine.split("=");
+            ElementLine line = readLine(rawLineSplit);
+            if (line != null) {
+                lines.add(line);
+            }
+        }
+        return lines;
+    }
```

Comment:
Isn't this a normal property file? If so, you can read it using `Properties` class from the JDK.
When you have the `Properties`, you can further parse them yourself - but the `Properties` class will do the underlying I/O and syntax checking for you.